<hr>
<b><font color='red'>Ejercicio 01</font></b>  

Lee el archivo.txt 

In [28]:
archivo="archivo.txt"
"""Esribe tu código aquí"""
f = open(archivo,"r")
array=f.read()
numeros=array.split()
counter=1;
for i in numeros:
    numero=float(i)
    if(counter%2!=0):
        print('{:.1f}'.format(numero),end='   ')
        counter+=1
    else:
        print('{:.2f}'.format(numero))
        counter+=1

1.0   2.00
2.0   4.00
3.0   6.00
4.0   8.00


y haz que tenga la siguiente salida:

    1.0   2.00  
    2.0   4.00  
    3.0   6.00  
    4.0   8.00  


<hr>
<b><font color='red'>Ejercicio 02</font></b>  

Haz lo mismo con el archvo archivo_error.txt,con cuidado de que a este archivo le falta algún dato, haz que en ese caso cuando lo lea, si no hay dos columnas no lo escriba.

In [69]:
"""Esribe tu código aquí"""
with open('archivo_error.txt', 'r') as archivo:
    lineas = archivo.readlines()
lineas_formateadas = []
for linea in lineas:
    numeros = linea.split()
    if len(numeros) >= 2:
        numero1_formateado = f"{float(numeros[0]):.1f}"
        numero2_formateado = f"{float(numeros[1]):.2f}"
        linea_formateada = f"{numero1_formateado}   {numero2_formateado}"
        lineas_formateadas.append(linea_formateada)
for linea in lineas_formateadas:
    print(linea)

1.0   2.00
3.0   6.00
4.0   8.00


<hr>
<b><font color='red'>Ejercicio 03</font></b>  

Lee el archivo.txt, haz que imprima las columnas en las filas y las filas en las columnas

In [11]:
"""Esribe tu código aquí"""
with open("archivo.txt", "r") as archivo:
    array = archivo.read().split()
pares = []
impares = []
for i in range(len(array)):
    if i % 2 == 0:
        pares.append(array[i])
    else:
        impares.append(array[i])
array_bidimensional = [pares, impares]
for fila in array_bidimensional:
    print(" ".join(fila))

1 2 3 4
2 4 6 8


<hr>
<b><font color='red'>Ejercicio 04</font></b>  

1. Escribe en un archivo la siguiente matriz:
$
\begin{equation}
\begin{pmatrix}
3 & 1 & 3\\
-1 & 1 & 0\\
-2 & 4 & 1
\end{pmatrix}
\end{equation} 
$
2. Lee el archivo
3. Calcula con los datos leidos el determinante de la matriz y lo saque por pantalla

In [18]:
f = open("Matriz.txt","w")
fila1=[3,1,3]
fila2=[-1,1,0]
fila3=[-2,4,1]
array_bidimensional = [fila1, fila2,fila3]
for fila in array_bidimensional:
    fila_str = " ".join([str(x) for x in fila])
    f.write(fila_str + "\n")
matriz = []
with open("Matriz.txt", "r") as archivo:
    lineas = archivo.readlines()
    for linea in lineas:
        elementos = [int(elemento) for elemento in linea.strip().split()]
        matriz.append(elementos)
a, b, c = matriz[0]
d, e, f = matriz[1]
g, h, i = matriz[2]
print((a*b*c)+(h*d*c)+(g*b*f)-((i*e*a)+(a*f*h)+(i*b*d)))

-5


<hr>
<b><font color='red'>Ejercicio 05</font></b>  

En el criptoanálisis, la técnica de análisis de frecuencias consiste en el aprovechamiento de estudios sobre la frecuencia de las letras o grupos de letras en los idiomas para poder establecer hipótesis para aprovecharlas para poder descifrar un texto cifrado sin tener la clave de descifrado (romper). Es un método típico para romper cifrados clásicos. 
Haz una anailisis de frecuencias para los 10 primeros càpitulos de el libro "el_quijote.txt"

In [41]:
with open("el_quijote.txt", "r", encoding="utf-8") as archivo:
    texto = archivo.read().lower()
conteo_letras = {}
abecedario = 'abcdefghijklmnopqrstuvwxyz'
counter=0
for letra in abecedario:
    conteo_letras[letra] = 0
for caracter in texto:
    if caracter in abecedario:
        conteo_letras[caracter] += 1
        counter+=1
print(counter)
for letra, cantidad in conteo_letras.items():
    print(f"{letra}: {'{:.3f}'.format(cantidad/counter)}")

94074
a: 0.127
b: 0.017
c: 0.037
d: 0.053
e: 0.138
f: 0.005
g: 0.011
h: 0.012
i: 0.053
j: 0.007
k: 0.000
l: 0.060
m: 0.026
n: 0.066
o: 0.099
p: 0.021
q: 0.019
r: 0.061
s: 0.073
t: 0.037
u: 0.048
v: 0.011
w: 0.000
x: 0.000
y: 0.015
z: 0.004


Salida:

    A   0.127
    B   0.017
    C   0.037
    D   0.053
    E   0.138
    F   0.005
    .
    .
    .

<hr>
<b><font color='red'>Ejercicio 06</font></b>  

haz un analisis de frecuencias en el libro encriptado "RH RQCEPESEFM AR MPM.txt", y comparala con el analisis anterior


In [54]:
abc="ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
archivoCodificar="RH RQCEPESEFM AR MPM.txt"

"""Esribe tu código aquí"""
with open("el_quijote.txt", "r", encoding="utf-8") as archivo:
    texto = archivo.read().lower()  
conteo_letras = {}
abecedario = 'abcdefghijklmnopqrstuvwxyz'
counter=0
for letra in alfabeto:
    conteo_letras[letra] = 0
for caracter in texto:
    if caracter in abecedario:
        conteo_letras[caracter] += 1
        counter+=1
with open(archivoCodificar, "r", encoding="utf-8") as archivo2:
    texto2 = archivo2.read().lower()
conteo_letras2 = {}
counter2=0
for letra in abecedario:
    conteo_letras2[letra] = 0
for caracter in texto2:
    if caracter in abecedario:
        conteo_letras2[caracter] += 1
        counter2+=1
print("el_quijote.txt   RH RQCEPESEFM AR MPM.txt")
for letra in abecedario:
    frecuencia_rel1 = conteo_letras[letra] / counter 
    frecuencia_rel2 = conteo_letras2[letra] / counter2 
    print(f"{letra}: {'{:.3f}'.format(frecuencia_rel1)}" f"           {letra}:{'{:.3f}'.format(frecuencia_rel2)}")

el_quijote.txt   RH RQCEPESEFM AR MPM.txt
a: 0.127           a:0.050
b: 0.017           b:0.006
c: 0.037           c:0.040
d: 0.053           d:0.061
e: 0.138           e:0.127
f: 0.005           f:0.007
g: 0.011           g:0.000
h: 0.012           h:0.053
i: 0.053           i:0.033
j: 0.007           j:0.012
k: 0.000           k:0.069
l: 0.060           l:0.001
m: 0.026           m:0.094
n: 0.066           n:0.027
o: 0.099           o:0.010
p: 0.021           p:0.067
q: 0.019           q:0.070
r: 0.061           r:0.137
s: 0.073           s:0.017
t: 0.037           t:0.045
u: 0.048           u:0.046
v: 0.011           v:0.009
w: 0.000           w:0.000
x: 0.000           x:0.002
y: 0.015           y:0.009
z: 0.004           z:0.004


salida:

    el_quijote.txt   RH RQCEPESEFM AR MPM.txt
    A   0.127        A   0.049
    B   0.017        B   0.006
    C   0.037        C   0.040
    .                .
    .                .
    .                .

<hr>
<b><font color='red'>Ejercicio 07</font></b>  

Comparando el analisis de frecuencia sabrías decir cual es el titulo del libro.  
RH RQCEPESEFM AR MPM

El Quijote